mmpose/datasets/datasets/animal/animal_pose_dataset.py

```diff
@@ -0,0 +1,370 @@
+
+    `Cross-domain Adaptation For Animal Pose Estimation’ ICCV'2019
+    More details can be found in the `paper
+    <https://arxiv.org/abs/1908.05806>`__ .
+
+    The dataset loads raw features and apply specified transforms
+    to return a dict containing the image tensors and other information.
+
+    Animal-Pose keypoint indexes::
+
+        0: 'L_Eye',
+        1: 'R_Eye',
+        2: 'L_EarBase',
+        3: 'R_EarBase',
+        4: 'Nose',
+        5: 'Throat',
+        6: 'TailBase',
+        7: 'Withers',
+        8: 'L_F_Elbow',
+        9: 'R_F_Elbow',
+        10: 'L_B_Elbow',
+        11: 'R_B_Elbow',
+        12: 'L_F_Knee',
+        13: 'R_F_Knee',
+        14: 'L_B_Knee',
+        15: 'R_B_Knee',
+        16: 'L_F_Paw',
+        17: 'R_F_Paw',
+        18: 'L_B_Paw',
+        19: 'R_B_Paw'
+
+    Args:
+        ann_file (str): Path to the annotation file.
+        img_prefix (str): Path to a directory where images are held.
+            Default: None.
+        data_cfg (dict): config
+        pipeline (list[dict | callable]): A sequence of data transforms.
+        test_mode (bool): Store True when building test or
+            validation dataset. Default: False.
+    """
+
+    def __init__(self,
+                 ann_file,
+                 img_prefix,
+                 data_cfg,
+                 pipeline,
+                 test_mode=False):
+        super().__init__(
+            ann_file, img_prefix, data_cfg, pipeline, test_mode=test_mode)
+
+        self.use_gt_bbox = data_cfg['use_gt_bbox']
+        self.bbox_file = data_cfg['bbox_file']
+        self.det_bbox_thr = data_cfg.get('det_bbox_thr', 0.0)
+        if 'image_thr' in data_cfg:
+            warnings.warn(
+                'image_thr is deprecated, '
+                'please use det_bbox_thr instead', DeprecationWarning)
+            self.det_bbox_thr = data_cfg['image_thr']
+        self.use_nms = data_cfg.get('use_nms', True)
+        self.soft_nms = data_cfg['soft_nms']
+        self.nms_thr = data_cfg['nms_thr']
+        self.oks_thr = data_cfg['oks_thr']
+        self.vis_thr = data_cfg['vis_thr']
+
+        self.ann_info['flip_pairs'] = [[0, 1], [2, 3], [8, 9], [10, 11],
+                                       [12, 13], [14, 15], [16, 17], [18, 19]]
+
+        self.ann_info['upper_body_ids'] = (0, 1, 2, 3, 4, 5, 7, 8, 9, 12, 13,
+                                           16, 17)
+        self.ann_info['lower_body_ids'] = (6, 10, 11, 14, 15, 18, 19)
+
+        self.ann_info['use_different_joint_weights'] = False
+        self.ann_info['joint_weights'] = np.array(
+            [
+                1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.2, 1.2, 1.2,
+                1.2, 1.5, 1.5, 1.5, 1.5
+            ],
+            dtype=np.float32).reshape((self.ann_info['num_joints'], 1))
+
+        # Note: The original paper did not provide enough information about
+        # the sigmas. We modified from 'https://github.com/cocodataset/'
```

Comment:
Can sigmas be computed from data?

Reply:
No. The sigmas measure the accuracy of the annotation, which should be provided by the dataset.

Reply:
Okay